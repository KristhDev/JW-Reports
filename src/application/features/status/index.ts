export { default as statusReducer, STATUS_INITIAL_STATE, clearStatus, setStatus } from './slice';
export * from './types';