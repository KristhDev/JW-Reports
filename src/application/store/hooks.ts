import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();