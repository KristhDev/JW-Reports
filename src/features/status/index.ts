export { default as statusReducer, clearStatus, setStatus } from './slice';