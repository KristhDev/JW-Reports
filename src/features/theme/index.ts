export { default as themeReducer, setTheme } from './slice';