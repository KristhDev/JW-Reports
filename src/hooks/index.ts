export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';