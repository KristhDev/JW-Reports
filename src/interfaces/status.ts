export interface StatusState {
    code: number;
    msg: string;
}

export type SetStatusPayload = {
    code: number;
    msg: string;
}