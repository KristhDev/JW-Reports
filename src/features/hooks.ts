import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;