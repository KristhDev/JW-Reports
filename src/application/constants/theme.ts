import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];