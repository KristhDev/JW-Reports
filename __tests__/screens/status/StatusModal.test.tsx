import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import hexToRgba from 'hex-to-rgba';

import { StatusModal } from '../../../src/screens/status';

import { useStatus, useTheme } from '../../../src/hooks';

import { darkColors } from '../../../src/theme';

const clearStatusMock = jest.fn();

jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <StatusModal /> screen', () => {
    (useStatus as jest.Mock).mockReturnValue({
        state: { msg: 'Test message of status modal.' },
        clearStatus: clearStatusMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
        BUTTON_TRANSPARENT_COLOR: hexToRgba(darkColors.button, 0.50)
    });

    beforeEach(() => {
        render(<StatusModal />);
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should call clearStatus when confirm button is pressed', () => {
        const touchable = screen.getAllByTestId('button-touchable')[0];
        fireEvent.press(touchable);

        expect(clearStatusMock).toHaveBeenCalled();
    });

    it('should render two buttons when espesific message', () => {
        (useStatus as jest.Mock).mockReturnValue({
            state: { msg: 'Para realizar est치 acci칩n necesitas permisos del dispositivo, por favor abra la configuraci칩n de su dispositivo y active los permisos de la aplicaci칩n.' },
            clearStatus: clearStatusMock
        });

        render(<StatusModal />);

        const buttons = screen.getAllByTestId('button-touchable');
        expect(buttons).toHaveLength(2);
    });
});