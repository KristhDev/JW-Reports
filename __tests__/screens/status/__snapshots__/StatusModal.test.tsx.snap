// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <StatusModal /> screen should to match snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 999,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#292929",
          "borderRadius": 10,
          "justifyContent": "space-between",
          "minHeight": 120,
          "padding": 15,
          "width": "87%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#B4B4B4",
              "fontSize": 19,
              "marginBottom": 10,
              "marginTop": 5,
              "width": "90%",
            }
          }
        >
          Test message of status modal.
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 20,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Array [
                Object {
                  "backgroundColor": "#C0A7E1",
                  "borderRadius": 10,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
            ]
          }
          testID="button-touchable"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "paddingVertical": 8,
                },
                Object {
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "color": "#121212",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginBottom": 2,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#C0A7E1",
                      "fontSize": 16,
                    },
                  ],
                ]
              }
              testID="button-text"
            >
              EST√Å BIEN
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
